src/Core/MailTemplate/Transformation/HTMLTextifyTransformation.php

```diff
@@ -0,0 +1,53 @@
+<?php
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\MailTemplate\Transformation;
+
+use Html2Text\Html2Text;
+
+class HTMLTextifyTransformation extends AbstractMailTemplateTransformation
+{
+    /**
+     * @inheritDoc
+     */
+    public function apply($templateContent, array $templateVariables)
+    {
+        $textifier = new Html2Text($templateContent);
+        $templateContent = $textifier->getText();
+
+        $templateContent = preg_replace('/^\s+/m', "", $templateContent);
```

Comment:
Why removing all whitespace characters?

Reply:
it's only the whitespaces at the start of line

Reply:
So the regex is wrong because:
1) it's not global
2) it will remove all \n

if you need to remove only whitespaces you need to do `/^ +/mg`

Reply:
no I want to remove `\t\r` and so on so `\s` seems appropriate, besides it won't remove all the `\n` thanks to the `m` modifier with targets line by line
I tested a few ones and this one gave me the cleaner output, besides I tried it but got an error when using the `g` modifier:

```
[Symfony\Component\Debug\Exception\ContextErrorException]
  Warning: preg_replace(): Unknown modifier 'g'
```

Reply:
modifier 'g' is implicit with preg_replace, did not know that :P
I misspoke for the `\n` sorry ^^

The problem is if you have:
```
Hi,

Nice to meet you,


From Mother Russia,
Kind regards,
```

you will have:
```
Hi,
Nice to meet you,
From Mother Russia,
Kind regards,
```
Which is weird for a txt mail.

Reply:
I changed the library to transform html to text It works much better, no need for additional `preg_replace`